tests/unit/test_compression.py

```diff
@@ -20,16 +20,16 @@ def test_compressarr():
 
 def test_compress_array_usesLZ4():
     cfn = Mock()
-    with patch('arctic._compression.clz4.compressarr', cfn):
+    with patch('arctic._compression.lz4_compress', cfn):
         compress_array([b"foo"] * 100)
-        assert cfn.call_count == 1
+        assert len(cfn.call_args_list) == 100  # call_count is not thread safe
```

Comment:
we've been biten by this intermittently failing test.
Mock's call count is not reliable with mutlithreading.
The increment is not a single Python vm instruction/atomic:
https://github.com/python/cpython/blob/master/Lib/unittest/mock.py#L957
and
https://stackoverflow.com/questions/39332139/thread-safe-version-of-mock-call-count

A trivial example:
```
import multiprocessing as mp
from mock import Mock

mymock = Mock()

def Process(n):
    for i in range(n):
        mymock.visit()

n_threads = 5
n_calls = 10000
result=mp.pool.ThreadPool(n_threads).map(Process, [n_calls]*n_threads)
print(mymock.visit.call_count)
assert mymock.visit.call_count == n_threads * n_calls
```

Reply:
```
>>> import dis
... def f(x):
...     x+=1
...     return x
... dis.dis(f)
  3           0 LOAD_FAST                0 (x)
              3 LOAD_CONST               1 (1)
              6 INPLACE_ADD         
              7 STORE_FAST               0 (x)

  4          10 LOAD_FAST                0 (x)
             13 RETURN_VALUE
```

 "i += 1" is not an atomic python Interpreter instruction
and GIL may be released in between the load and the INPLACE_ADD/STORE_FAST